pkg/volume/flexvolume/util.go

```diff
@@ -104,7 +104,7 @@ func getReadOnly(spec *volume.Spec) (bool, error) {
 	}
 	if spec.PersistentVolume != nil && spec.PersistentVolume.Spec.FlexVolume != nil {
 		// ReadOnly is specified at the PV level
-		return spec.ReadOnly, nil
+		return spec.PersistentVolume.Spec.FlexVolume.ReadOnly, nil
```

Comment:
I think we should combine both `spec.ReadOnly` and `spec.PersistentVolume.Spec.FlexVolume.ReadOnly`, as `spec.ReadOnly` comes from `Pod.Spec.Volumes.VolumeSource.PersistentVolumeClaim.ReadOnly`.